src/cpu/Core.cpp

```diff
@@ -43,13 +43,13 @@ void Core::instr_ldhl(Byte n) {
 // ----------------------------------------------------------------------------
 
 void Core::instr_push(Word v) {
-  write(--_sp.word, static_cast<Byte>((v & 0xFF00) >> 8));
-  write(--_sp.word, static_cast<Byte>((v & 0x00FF)));
+  _components.mem_bus->write(--_sp.word, static_cast<Byte>((v & 0xFF00) >> 8));
+  _components.mem_bus->write(--_sp.word, static_cast<Byte>((v & 0x00FF)));
```

Comment:
This just delays the bug. If the stack pointer is invalid, this function will corrupt the memory instead of aborting the program (the assert in core::write does that).
For me, push and pop should NOT write outside the core stack!! Thus using core::write(); should not be an issue at all.